src/components/dashboard/Dashboard.jsx

```diff
@@ -35,41 +36,40 @@ const Dashboard = (props) => {
     _.isEmpty(historicalResources)
 
   return (
-    <section id="dashboard">
-      <Header triggerEditorMenu={props.triggerHandleOffsetMenu} />
-      <Alerts errorKey={dashboardErrorKey} />
-      <PreviewModal errorKey={dashboardErrorKey} />
-      {showWelcome && (
-        <div>
-          <h2>Welcome to Sinopia.</h2>
-          <p>
-            As you use Sinopia, your most recently used templates, resources,
-            and searches will appear on this dashboard.
-          </p>
-        </div>
-      )}
-      {!_.isEmpty(historicalTemplates) && (
-        <div>
-          <h2>Recent templates</h2>
-          <ResourceTemplateSearchResult
-            results={historicalTemplates}
-            errorKey={dashboardErrorKey}
-          />
-        </div>
-      )}
-      {!_.isEmpty(historicalSearches) && (
-        <div>
-          <h2>Recent searches</h2>
-          <SearchList searches={historicalSearches} />
-        </div>
-      )}
-      {!_.isEmpty(historicalResources) && (
-        <div>
-          <h2>Recent resources</h2>
-          <ResourceList resources={historicalResources} />
-        </div>
-      )}
-    </section>
+    <AlertsContextProvider value={dashboardErrorKey}>
```

Comment:
Each page (like Dashboard, Search, Editor, Templates, Exports, etc.) are going to be wrapped in a AlertsContextProvider and then have a ContextAlert (below) to render the relevant alerts.